### PR TITLE
add ssh infra driver

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

Through the re-encapsulation of the ssh package, the cluster operation can be completed directly through the ip address, instead of passing the v2.cluster structure to the ssh inner layer as before we do.

### CODE REVIEW

File: pkg/infradriver/ssh_infradriver.go [hunk 1]
```diff
@@ -1,56 +1,143 @@
...
+		}
+		for _, ip := range host.IPS {
+			ret.sshConfigs[ip.String()] = ssh.NewSSHClient(&host.SSH, true)
+		}
+	}
+
+	// initialize roleHostsMap field
+	for _, host := range cluster.Spec.Hosts {
+		for _, role := range host.Roles {
+			ret.roleHostsMap[role] = host.IPS
```
Comment (VinceCui) [inline pkg/infradriver/ssh_infradriver.go:53]: need append, because different host maybe have same role.

Comment (VinceCui) [reply pkg/infradriver/ssh_infradriver.go]: Any update？

Comment (kakaZhou719) [reply pkg/infradriver/ssh_infradriver.go]: it already fixed, pls review.

File: pkg/infradriver/ssh_infradriver.go [hunk 2]
```diff
@@ -1,56 +1,167 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 package infradriver
 
 import (
+	"fmt"
+	"github.com/imdario/mergo"
```
Comment (VinceCui) [inline pkg/infradriver/ssh_infradriver.go:19]: group them

**Review by VinceCui — APPROVED** [review]
LGTM

**Pull request merged**